File: codewars-ts/src/app/codewars/CarMileage.ts
export function isInteresting(n: number, awesomePhrases: number[]): number {
    if ( isInterestingImpl(n, awesomePhrases) ) {
        return 2;
    }
    for(let i=1; i<=2; i++) {
        if ( isInterestingImpl(n + i, awesomePhrases) ) {
            return 1;
        }
    }
    return 0;
}

function isInterestingImpl(n: number, awesomePhrases: number[]): boolean {
    if ( awesomePhrases.indexOf(n) >= 0 ) {
        return true;
    }
    const s = n.toString();
    return s.length >= 3 && (
        /^[1-9]0+$/.test(s) || 
        /^([1-9])\1+$/.test(s) ||
        '1234567890'.indexOf(s) >= 0 || 
        '9876543210'.indexOf(s) >= 0 || 
        s === s.split("").reverse().join(""));
}

File: codewars-java/main/java/com/acebanenco/codewars/CarMileage.java
package com.acebanenco.codewars;

import java.util.Arrays;

public class CarMileage {

    public static int isInteresting(int number, int[] awesomePhrases) {
        CarMileage carMileage = new CarMileage();
        return carMileage.isInterestingImpl(number, awesomePhrases);
    }

    int isInterestingImpl(int number, int[] awesomePhrases) {
        if ( testInteresting(number, awesomePhrases) ) {
            return 2;
        }
        for (int mile = 1; mile <= 2; mile++) {
            if (testInteresting(number + mile, awesomePhrases) ) {
                return 1;
            }
        }
        return 0;
    }

    boolean testInteresting(int number, int[] awesomePhrases) {
        if ( isAwesome(number, awesomePhrases) ) {
            return true;
        }
        String input = "" + number;
        if ( input.length() < 3 ) {
            return false;
        }
        if ( isDigitFollowedByZero(input) ) {
            return true;
        }
        if ( isEveryDigitSame(input) ) {
            return true;
        }
        if ( isSequentialIncrementing(input) ) {
            return true;
        }
        if ( isSequentialDecrementing(input) ) {
            return true;
        }
        return isPalindrome(input);
    }

    boolean isAwesome(int number, int[] awesomePhrases) {
        return Arrays.stream(awesomePhrases)
                .anyMatch(n -> n == number);
    }

    boolean isDigitFollowedByZero(String input) {
        return input.matches("^[1-9]0+$");
    }

    boolean isEveryDigitSame(String input) {
        return input.matches("^([1-9])\\1+$");
    }

    boolean isSequentialDecrementing(String input) {
        return "9876543210".contains(input);
    }

    boolean isSequentialIncrementing(String input) {
        return "1234567890".contains(input);
    }

    boolean isPalindrome(String input) {
        return new StringBuilder(input)
                .reverse()
                .toString()
                .equals(input);
    }

}
